sdk/eventhub/azure-eventhubs/azure/eventhub/producer.py

```diff
@@ -118,39 +117,30 @@ def _open(self, timeout_time=None):
             self.target = self.redirected.address
         super(EventHubProducer, self)._open(timeout_time)
 
-    def _send_event_data(self, timeout=None):
-        timeout = timeout or self.client.config.send_timeout
-        if not timeout:
-            timeout = 100000  # timeout None or 0 mean no timeout. 100000 seconds is equivalent to no timeout
-        start_time = time.time()
-        timeout_time = start_time + timeout
-        max_retries = self.client.config.max_retries
-        retry_count = 0
-        last_exception = None
-        while True:
-            try:
-                if self.unsent_events:
-                    self._open(timeout_time)
-                    remaining_time = timeout_time - time.time()
-                    if remaining_time <= 0.0:
-                        if last_exception:
-                            error = last_exception
-                        else:
-                            error = OperationTimeoutError("send operation timed out")
-                        log.info("%r send operation timed out. (%r)", self.name, error)
-                        raise error
-                    self._handler._msg_timeout = remaining_time  # pylint: disable=protected-access
-                    self._handler.queue_message(*self.unsent_events)
-                    self._handler.wait()
-                    self.unsent_events = self._handler.pending_messages
-                    if self._outcome != constants.MessageSendResult.Ok:
-                        if self._outcome == constants.MessageSendResult.Timeout:
-                            self._condition = OperationTimeoutError("send operation timed out")
-                        _error(self._outcome, self._condition)
-                return
-            except Exception as exception:
-                last_exception = self._handle_exception(exception, retry_count, max_retries, timeout_time)
-                retry_count += 1
+    @_retry_decorator
+    def _send_event_data(self, **kwargs):
```

Comment:
No need to use `**kwargs` for two positional parameters for an internal method.

Reply:
It is fixed.